src/main/target/common_post.h

```diff
@@ -353,6 +353,10 @@ extern uint8_t __config_start;   // configured via linker script when building b
 extern uint8_t __config_end;
 #endif
 
+#if defined(EEPROM_IN_RAM)
```

Comment:
this condition is not correct, there should be nothing stopping a developer compiling any target with EEPROM_IN_RAM *without* using EXST.

Reply:
the build system should add USE_EXST (via the target's `EXST=yes` flag in the make script).  It's not valid to build a EXST firmware without the EXST block or linker script.  This will likely confuse anyone looking at the EXST system.

Suggest renaming the target from NUCLEOH743_RAMCONFIG to NUCLEOH743EXST and then adding a NUCLEOH743EXST.mk file, which enabled EXST=yes and then conditionally exclude some features so the target will fit into available flash space.

finally, use an H743 exst linker script that defines the .exst_hash block, refer to the H750 exst linker script for details.

Reply:
My bad, I got confused. Maybe we should rename `EEPROM_IN_RAM` to `CONFIG_IN_RAM` or `EEPROM_CONFIG_IN_RAM` to make it more explicit that this only applies to the config?

Reply:
> Suggest renaming the target from NUCLEOH743_RAMCONFIG to NUCLEOH743EXST and then adding a NUCLEOH743EXST.mk file, which enabled EXST=yes and then conditionally exclude some features so the target will fit into available flash space.

Ok, here is the bit where I don't quite get it: Doesn't `EXST` mean that the target is built so that it would be written into *external* flash space? So where does the 'available flash space' limitation come from? Or is that meant to mean the available *RAM*?

Reply:
Yes, the `EEPROM_IN_*` should be renamed to `CONFIG_IN_*` at some point, likely when an interface is extracted from the existing config/eeprom storage systems, but not now.

Yes, EXST means the code is written to external flash space 'EXST = External Storage'.

Yes, when I said 'available flash space' I should have said 'available code space' which means RAM for an EXST target.

Reply:
@hydra: So this is all good to go now then?

Reply:
if it does what you want it to, yes, but see my concerns with this PR on the main comment thread.